src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.cpp

```diff
@@ -0,0 +1,384 @@
+    }
+}
+
+CHIP_ERROR ServerInstance::ReadFeatureMap(BitFlags<Feature> & featureMap)
+{
+    bool panChangeSupported;
+    ReturnErrorOnFailure(mDelegate->GetPanChangeSupported(panChangeSupported));
+    if (panChangeSupported)
+    {
+        featureMap.Set(Feature::kPANChange);
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadBorderRouterName(MutableCharSpan & borderRouterName)
+{
+    ReturnErrorOnFailure(mDelegate->GetBorderRouterName(borderRouterName));
+    ReturnErrorCodeIf(borderRouterName.size() > kBorderRouterNameMaxLength, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadBorderAgentID(MutableByteSpan & borderAgentId)
+{
+    ReturnErrorOnFailure(mDelegate->GetBorderAgentId(borderAgentId));
+    ReturnErrorCodeIf(borderAgentId.size() != kBorderAgentIdLength, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadThreadVersion(uint16_t & threadVersion)
+{
+    return mDelegate->GetThreadVersion(threadVersion);
+}
+
+CHIP_ERROR ServerInstance::ReadInterfaceEnabled(bool & interfaceEnabled)
+{
+    return mDelegate->GetInterfaceEnabled(interfaceEnabled);
+}
+
+CHIP_ERROR ServerInstance::ReadActiveDatasetTimestamp(Optional<uint64_t> & activeDatasetTimestamp)
+{
+    uint64_t activeDatasetTimestampValue;
+    Thread::OperationalDataset activeDataset;
+    if (mDelegate->GetDataset(activeDataset, Delegate::DatasetType::kActive) == CHIP_NO_ERROR &&
+        activeDataset.GetActiveTimestamp(activeDatasetTimestampValue) == CHIP_NO_ERROR)
+    {
+        activeDatasetTimestamp.SetValue(activeDatasetTimestampValue);
+    }
+    else
+    {
+        activeDatasetTimestamp.ClearValue();
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    if (aPath.mClusterId != ThreadBorderRouterManagement::Id)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    if (!mDelegate)
+    {
+        ChipLogError(Zcl, "Thread Border Router Management server not initialized");
+        return CHIP_ERROR_UNINITIALIZED;
+    }
+    CHIP_ERROR status = CHIP_NO_ERROR;
+    switch (aPath.mAttributeId)
+    {
+    case Globals::Attributes::FeatureMap::Id: {
+        BitFlags<Feature> featureMap;
+        status = ReadFeatureMap(featureMap);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(featureMap);
+        }
+        break;
+    }
+    case Attributes::BorderRouterName::Id: {
+        char borderRouterNameBuf[kBorderRouterNameMaxLength + 1];
+        MutableCharSpan borderRouterName(borderRouterNameBuf);
+        status = ReadBorderRouterName(borderRouterName);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(borderRouterName);
+        }
+        break;
+    }
+    case Attributes::BorderAgentID::Id: {
+        uint8_t borderAgentIDBuf[kBorderAgentIdLength];
```

Comment:
Initialize all stack variables to good defaults.

Reply:
Done